assets/src/stories-editor/components/higher-order/with-replaced-full-size-image.js

```diff
@@ -0,0 +1,66 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { dispatch, select } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { MAX_IMAGE_SIZE_SLUG } from '../../constants';
+
+/**
+ * A component that extends the BlockEdit components, replacing the 'full' image size with a custom max size.
+ *
+ * @param {Function} BlockEdit A block's edit component to be extended.
+ * @return {Function} A function returning an extended BlockEdit.
+ */
+export default ( BlockEdit ) => {
+	return class ReplacedFullSizeImage extends BlockEdit {
+		/**
+		 * The class constructor.
+		 */
+		constructor() {
+			super( ...arguments );
+			this.componentDidUpdate = this.componentDidUpdate.bind( this );
+			this.ampReplaceFullSizeImage = this.ampReplaceFullSizeImage.bind( this );
+		}
+
+		/**
+		 * When the component updates, if it's for the Image block, replace the full size image.
+		 */
+		componentDidUpdate() {
```

Comment:
Is it not possible to intercept the media upon selection to replace the `full` with the desired image size? Then there wouldn't be a need to turnaround and update it again. Or maybe I am not understanding enough.

Reply:
Good question. That entire extending JS class isn't a great approach.

https://github.com/ampproject/amp-wp/pull/2509/commits/236e2265c2cbfb442ae5f6670245df9f64ddda17 removes that entire JS class above, in favor of simply [removing the full size](https://github.com/ampproject/amp-wp/pull/2509/commits/236e2265c2cbfb442ae5f6670245df9f64ddda17#diff-37ed1ba86eaee9e72c2ced620309452bR1532) in the filter that you recommended.